capstone/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -20,4 +31,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("text/html;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    String founderEmail = userService.getCurrentUser().getEmail();
+
+    String clubName = request.getParameter("name");
+    String description = request.getParameter("description");
+    String website = request.getParameter("website");
+    BlobKey key = getBlobKey(request, "logo");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity clubEntity = new Entity("Club", clubName);
+    clubEntity.setProperty("name", clubName);
+    clubEntity.setProperty("description", description);
+    clubEntity.setProperty("members", ImmutableList.of(founderEmail));
+    clubEntity.setProperty("officers", ImmutableList.of(founderEmail));
+    clubEntity.setProperty("announcements", ImmutableList.of(""));
+    clubEntity.setProperty("logo", key);
+    datastore.put(clubEntity);
+  }
+
+  /* Return BlobKey for image uploaded through form. */
+  private BlobKey getBlobKey(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
```

Comment:
Does returning null lead to a nullpointerexception? If so, maybe we can return the URL to a default image or something here?

Reply:
I don't think it will lead to an exception, but this is a good point. But I think since a default image would be static/constant, I think i might just check if the blobkey prop is null and handle it on that end, instead of using blobstore to store a default image.